plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -329,6 +329,27 @@ func ClusterRoles() []rbacv1.ClusterRole {
 				rbacv1helpers.NewRule(Read...).Groups(networkingGroup).Resources("networkpolicies").RuleOrDie(),
 			},
 		},
+		{
+			// a role for namespace level metrics viewing (aggregated to view).  It grants read
+			// access to resource metrics on pods and custom metrics on all the normal view resources.
+			// We explicitly skip external metris, since some external metrics pipelines ignore the
+			// namespace component, and in thus could be used to gain access to metrics outside the
+			// appropriate namespace).
+			ObjectMeta: metav1.ObjectMeta{Name: "system:namespaced-metrics-reader", Labels: map[string]string{"rbac.authorization.k8s.io/aggregate-to-view": "true"}},
+			Rules: []rbacv1.PolicyRule{
+				rbacv1helpers.NewRule("get", "list").Groups(resMetricsGroup).Resources("pods").RuleOrDie(),
+				rbacv1helpers.NewRule("get", "list").Groups(customMetricsGroup).Resources("pods", "replicationcontrollers", "serviceaccounts",
```

Comment:
is there no way to say "all" here?

Reply:
all is not the right verb -- you don't want people to be able to list metrics for objects that they can't otherwise list (or, at least, not necessarily).